SignalServiceKit/src/Util/OWSAnalytics.m

```diff
@@ -145,46 +162,103 @@ - (void)tryToSyncEvents
             if (!firstEventKey) {
                 return;
             }
-
+            
             firstEventDictionary = [transaction objectForKey:firstEventKey inCollection:kOWSAnalytics_EventsCollection];
             OWSAssert(firstEventDictionary);
             OWSAssert([firstEventDictionary isKindOfClass:[NSDictionary class]]);
         }];
 
-        if (!firstEventDictionary) {
-            return;
+        if (firstEventDictionary) {
+            [self sendEvent:firstEventDictionary eventKey:firstEventKey isCritical:NO];
         }
+    });
+}
 
-        DDLogDebug(@"%@ trying to deliver event: %@", self.tag, firstEventKey);
+- (void)sendEvent:(NSDictionary *)eventDictionary eventKey:(NSString *)eventKey isCritical:(BOOL)isCritical
+{
+    OWSAssert(eventDictionary);
+    OWSAssert(eventKey);
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
+    OWSAssert(dispatch_get_current_queue() == self.serialQueue);
```

Comment:
We achiever a similar end through different means in PeerConnectionClient and SessionCipher. It would be nice if we had just one approach to this. I understand the drawbacks in the approach used elesewhere to be:

1. only runs on iOS10+
2. Since check actually crashes the app, we only run it in debug, meaning no sign of assertion failure in Production.

I'm less clear on the drawbacks to this approach based on the reasons that this method was deprecated - do we always get the correct answer here?

Anyway, it'd be nice to centralize on the "one true way" and use it consistently.